syntax = "proto3";

// ZirconiumBank (ZrB)

/* The User can requestt for the following from the server
 * 1. Withdraw Amount
 * 2. Desposit Amount
 * 3. Balance Enquiry
 * 4. Change Pin // implement this
 * */

/* The Request should always contain
 * 1. Username
 * 2. 6-digit PIN
 * */

 // The C++ namesapce will be zirconium::bank
package zirconium.bank;

// Interface exported by the server
service WebService {
    rpc WithdrawAmount(RequestAmount) returns (Balance) {}
    rpc DepositAmount(RequestAmount) returns (Balance) {}
    rpc BalanceEnquiry(AuthUser) returns (Balance) {}
}

message AuthUser {
    string username = 1;
    string pin = 2;
}

message RequestAmount {
    AuthUser user = 1;
    int32 value = 2;
}

message Balance {
    int32 value = 1;
}
